pandas/core/arrays/numpy_.py

```diff
@@ -164,6 +164,10 @@ def _from_sequence(cls, scalars, dtype=None, copy=False):
             result = result.copy()
         return cls(result)
 
+    def diff(self, periods: int = 1):
+        result = algorithms.diff(com.values_from_object(self._ndarray), periods)
```

Comment:
`com.values_from_object` was the old behavior of Series.diff. So this should be identical to the old behavior, just with an extra function call to go through `self.array.diff`.

Reply:
Why do you need to override the base class version? Doesn't that work as well for numpy?